pyanaconda/storage/osinstall.py

```diff
@@ -82,6 +82,7 @@ def enable_installer_mode():
     # into the *host's* /etc/lvm. This can get real messy on build systems.
     if blivet_flags.image_install:
         blivet_flags.lvm_metadata_backup = False
+        blivet_flags.protect_cdroms = True
```

Comment:
I think that the code for protecting cdroms should be in anaconda. It could be probably done the same way as protecting zRAM swap devices: https://github.com/rhinstaller/anaconda/blob/rhel-devel/pyanaconda/anaconda.py#L177

Reply:
That wouldn't protect the device from teardown_all() that is called by

storage_initialize -> InstallerStorage.reset() -> Blivet.reset() -> DeviceTree.populate() which is calling DeviceTree.teardown_all().

The specs from anaconda.protected are applied after reset/populate (_mark_protected_device(dev)).

I didn't find a way how to prevent teardown of (just populated) cdrom in Blivet.reset() outside blivet.

Reply:
In that case, I would like to be able to propagate the list of protected devices to Blivet, so they can be skipped during the teardown, or the teardown shouldn't be part of the populate method and we can call it in Anaconda in our reset method. It seems weird to me that it is possible to teardown devices that should be protected. @vojtechtrefny, what do you think?

Reply:
@rvykydal did you tried if the CDROM installation works with this code. I fear that this could block us from installing from the media. Ideally also test if you are able to install from a different cdrom than you booted.

Reply:
@jkonecny12 The patch is limited to image installs. But I'll check if it does not break installing from cdrom.

Reply:
Actually we shouldn't differentiate by target (image) but system (system) in this case though - but that would be  for upstream. I think for rhel branch we just should go with flags.imageInstall.